test/public.spec.js

```diff
@@ -0,0 +1,423 @@
+    });
+
+    it('should return medium coverage color', function() {
+      expect(window.NCC.imageChecker._getBackgroundColor(150)).toEqual('orange');
+    });
+
+    it('should return low coverage color', function() {
+      expect(window.NCC.imageChecker._getBackgroundColor(75)).toEqual('green');
+    });
+
+    it('should return very low coverage color', function() {
+      expect(window.NCC.imageChecker._getBackgroundColor(74)).toEqual('blue');
+    });
+  });
+
+  function createBackgroundImg() {
+    backgroundImg = document.createElement('div');
+    backgroundImg.style = 'display: block;width: 200px;height: 160px; background: url("base/test/assets/placeholder-100x80.png");';
+    return backgroundImg;
+  }
+
+  function createNoBackgroundImg() {
+    noBackgroundImg = document.createElement('div');
+    return noBackgroundImg;
+  }
+
+  function createHiddenImg() {
+    hiddenImg = document.createElement('img');
+    hiddenImg.src = 'base/test/assets/placeholder-100x80.png';
+    hiddenImg.style = 'display: none;';
+    return hiddenImg;
+  }
+
+  function createInvisibleImg() {
+    invisibleImg = document.createElement('img');
+    invisibleImg.src = 'base/test/assets/placeholder-100x80.png';
+    invisibleImg.style = 'visibility: hidden;';
+    return invisibleImg;
+  }
+
+  function createNoSrcImg() {
+    noSrcImg = document.createElement('img');
+    return noSrcImg;
+  }
+
+  function createNoDimensionImg() {
+    noDimensionsImg = document.createElement('img');
+    noDimensionsImg.src = 'base/test/assets/placeholder-100x80.png';
+    noDimensionsImg.style = 'display: block; width: 0px; height: 0px;';
+    return noDimensionsImg;
+  }
+
+  function createSmallImg() {
+    smallWidthImg = document.createElement('img');
+    smallWidthImg.src = 'base/test/assets/placeholder-100x80.png';
+    smallWidthImg.style = 'display: block; width: 150px; height: 50px;';
+    return smallWidthImg;
+  }
+
+  function createMediumImg() {
+    smallWidthUrlImg = document.createElement('img');
+    smallWidthUrlImg.src = 'base/test/assets/placeholder-100x80.png';
+    smallWidthUrlImg.style = 'display: block; width: 151px; height: 71px;';
+    return smallWidthUrlImg;
+  }
+
+  function createImg() {
+    img = document.createElement('img');
+    img.src = 'base/test/assets/placeholder-100x80.png';
+    return img;
+  }
+
+  function createBigImg() {
+    bigImg = document.createElement('img');
+    bigImg.src = 'base/test/assets/placeholder-100x80.png';
+    bigImg.style = 'display: block;width: 200px;height: 160px;';
+    return bigImg;
+  }
+
+  function createMissingImg() {
+    missingImg = document.createElement('img');
+    missingImg.src = 'not-found.png';
+    return missingImg;
+  }
+
+  function createCrossDomainImg() {
+    crossDomainImg = document.createElement('img');
+    crossDomainImg.src = 'https://upload.wikimedia.org/wikipedia/commons/7/73/Flag_of_Romania.svg';
```

Comment:
@tabanliviu DoD - In all seriousness, this should probably be an asset owned by NCC.